src/components/Messages/index.js

```diff
@@ -1,20 +1,36 @@
-import { h, Component } from 'preact';
-import Message from 'components/Message';
-import { getAvatarUrl, getAttachmentsUrl } from '../helpers';
+import Message from '../Message';
+import TypingIndicator from '../TypingIndicator';
+import { createClassName, getAttachmentsUrl } from '../helpers';
+import styles from './styles';
 
-export default class Messages extends Component {
 
-	render({ messages, user, ...args }) {
-		return (
-			<ol style="padding:0;">{
-				messages.map((message, index, arr) => {
-					const next = arr[index + 1];
-					const { u = {}, _id, attachments } = message;
-					const group = next && next.u._id === u._id;
-					const username = u._id && u.username;
-					return <Message {...message} group={group} key={_id} el="li" me={user && user._id === u._id} avatarUrl={getAvatarUrl(username)} attachmentsUrl={getAttachmentsUrl(attachments)} />;
-				})
-			}</ol>
-		)
-	}
-}
+export const Messages = ({
+	user: { _id: userId, avatar: userAvatar = {} } = {},
+	agent: { _id: agentId, avatar: agentAvatar = {} } = {},
+	messages = [],
+	typingAvatars = [],
+}) => (
+	<ol className={createClassName(styles, 'messages')}>
+		{messages.map((message, index, arr) => {
+			const { _id: messageId, u: { _id: messageUserId } = {}, attachments } = message;
+			const { u: { _id: nextMessageUserId } = {} } = arr[index + 1] || {};
+
+			return (
+				<Message
+					el="li"
+					key={messageId}
+					me={userId === messageUserId}
```

Comment:
We have a problem here.. When the message was sent by a trigger, it is possible that both `userId` and `messageUserId` are equal to `undefined`. With the behaviour above, it will reflect just as the message was sent by the **user** and not the **agent**.